orthoinference/src/main/java/org/reactome/orthoinference/CandidateSetsConvertedToDefinedSetsSpeciesFix.java

```diff
@@ -0,0 +1,78 @@
+package org.reactome.orthoinference;
+
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.sql.SQLException;
+import java.util.Collection;
+import java.util.Properties;
+
+import org.gk.model.GKInstance;
+import org.gk.model.ReactomeJavaConstants;
+import org.gk.persistence.MySQLAdaptor;
+import org.reactome.release.common.database.InstanceEditUtils;
+
+/*
+ * This data fix was written to correct a mistake in the first version of the Java Orthoinference code.
+ * 
+ * During orthoinference, when CandidateSet instances contain multiple instances in the 'hasMember' attribute slot the inferred instance is changed
+ * from a CandidateSet to DefinedSet instance. At the location of the code in question (OrthologousEntity.createInfEntitySet in Java Orthoinference),
+ * the new instance is produced by creating a new GKInstance from a DefinedSet SchemaClass, rather than using the InstanceUtilities.newInferredGKInstance
+ * method that would normally be used (using that method would create a new CandidateSet instance, based on the source instance).
+ * 
+ * The mistake was that the 'species' and 'created' slots that are populated in 'newInferredGKInstance' were not populated when the new DefinedSet was created.
+ * This code fixes that mistake (Justin Cook November 2018).
+ * 
+ */
+
+public class CandidateSetsConvertedToDefinedSetsSpeciesFix {
```

Comment:
I think it might be better if one-time data fix code is maintained in a separate directory or project within the repository. The way it is right now, any future builds of the OrthoInference application will include this code, and it is completely unnecessary for them to do so. As an example, the KEGGIdentifierFix was moved to a separate project in the AddLinks repository (https://github.com/reactome/add-links/tree/develop/KeggIdentifierFixer). I probably could have also moved that into a "datafix" subdirectory to distinguish it from the main project, and I probably _will_ do that once there is more than a couple of data fix projects.

Reply:
Fair enough, I wanted it to be visible for transparency's sake but I'll move it to something akin to a 'datafix' directory

Reply:
Visibility is a very good thing, so thank you for doing that! :)

We just need to keep the code organized a little better than was done in the past. I tried to clean up the current Release repo by also adding a "fixes" directory for one-time fixer scripts (https://github.com/reactome/Release/tree/repository_cleanup/scripts/fixes) but by the time I finish the work on that cleanup effort, that whole repository might be mostly useless.